src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -313,6 +316,17 @@ public function checkAnnotations(\ReflectionClass $refl, $class)
             return $deprecations;
         }
 
+        if (
+            isset($parentAndOwnInterfaces['Serializable']) &&
+            0 === strpos($refl->getNamespaceName(), 'Symfony\\') &&
+            (
+                (!$refl->hasMethod('__serialize') || !($methodRefl = $refl->getMethod('__serialize'))->isPublic() || $methodRefl->isStatic() || !$refl->hasMethod('__unserialize') || !($methodRefl = $refl->getMethod('__unserialize'))->isPublic() || $methodRefl->isStatic()) &&
+                (!($isInterface = $refl->isInterface()) || ((self::$method[$class]['__serialize'][2] ?? true || self::$method[$class]['__unserialize'][2] ?? true)))
+            )
+        ) {
+            $deprecations[] = sprintf('The "%s" %s the broken "\Serializable" interface. It is discouraged to do so. It is going to be deprecated and removed in future PHP versions.', $class, $isInterface ? 'interface extends' : 'class implements');
```

Comment:
`The [..] the broken "Serializable" interface. You should implement the new "__serialize" and "__unserialize"  methods instead.`